depth-test_z-acc_spatial-noise/depth_test.py

```diff
@@ -0,0 +1,174 @@
+		(k_x, k_y, k_z), resid, rank, s = np.linalg.lstsq(G, Z, rcond=None)
+		normal = np.array([k_x, k_y, -1])
+		normal = normal / np.linalg.norm(normal)
+
+		self.plane_normal = normal
+		self.plane_distance = k_z
+		self.plane_coeffs = (k_x, k_y, k_z)
+
+		self.fitted = True
+
+		return normal, k_z
+
+	def compute_tilt(self):
+		horizontal_tilt = angle(self.camera_dir, self.plane_normal, np.array([0, 1, 0])) # in radians (- left, + right)
+		vertical_tilt = angle(self.camera_dir, self.plane_normal, np.array([1, 0, 0])) # in radians (- up, + down)
+
+		return horizontal_tilt, vertical_tilt
+
+	def print_tilt(self):
+		horizontal_tilt, vertical_tilt = self.compute_tilt()
+		print(
+			f"Horizontal tilt: {abs(horizontal_tilt) * 180/np.pi}° {'LEFT' if horizontal_tilt < 0 else 'RIGHT'}", 
+			f"Vertical tilt: {abs(vertical_tilt) * 180/np.pi}° {'UP' if vertical_tilt < 0 else 'DOWN'}"
+		)
+
+	def project_on_plane(self, point_cloud: o3d.geometry.PointCloud):
+		points = np.asarray(point_cloud.points)
+		G = np.ones_like(points)
+		G[:, 0] = points[:, 0]
+		G[:, 1] = points[:, 1]
+		plane_fit = o3d.geometry.PointCloud(o3d.utility.Vector3dVector(np.hstack([points[:,0:2], G @ np.matrix(self.plane_coeffs).T])))
+		return plane_fit
+
+	def correct_tilt(self, point_cloud: o3d.geometry.PointCloud):
+		# Correct the point cloud tilt (isolating camera positioning errors)
+		# The plane is corrected with angle (acos(camera_dir @ normal)) and axis (np.cross(camera_dir, normal)) instad of the horizontal and vertical tilt to reduce numerical errors
+		# R = o3d.geometry.get_rotation_matrix_from_xyz(np.array([-vertical_tilt, -horizontal_tilt, 0]))
+		axis_angle = normalize(np.cross(self.plane_normal, self.camera_dir)) * np.arccos(np.dot(self.camera_dir, self.plane_normal))
+		R = o3d.geometry.get_rotation_matrix_from_axis_angle(axis_angle)
+		point_cloud_corrected = o3d.geometry.PointCloud(point_cloud)
+		point_cloud_corrected.rotate(R, center=(0, 0, 0))
+		return point_cloud_corrected
+
+	def measure(self, point_cloud: o3d.geometry.PointCloud):
+		print(f"\rAdding measurements {'.'*self.samples}", end="")
+
+		plane_fit = self.project_on_plane(point_cloud)
+		plane_fit_corrected = self.correct_tilt(plane_fit)
+		point_cloud_corrected = self.correct_tilt(point_cloud)
+
+		spatial_noise = self.compute_spatial_noise(point_cloud_corrected, plane_fit_corrected)
+		z_accuracy = self.compute_z_accuracy(point_cloud_corrected)
+
+		self.z_accuracy_medians.append(z_accuracy)
+		self.spatial_noise_rmses.append(spatial_noise)
+		self.samples += 1
+
+		return spatial_noise, z_accuracy
+
+	def compute_spatial_noise(self, point_cloud_corrected: o3d.geometry.PointCloud, plane_fit_corrected: o3d.geometry.PointCloud):
+		plane_fit_corrected_points = np.asarray(plane_fit_corrected.points)
+		point_cloud_corrected_points = np.asarray(point_cloud_corrected.points)
+
+		z_error = plane_fit_corrected_points[:,2] - point_cloud_corrected_points[:,2]
+		z_error = np.sort(z_error)
```

Comment:
Use disparity here. Considering plane estimatin is correct. 
![image](https://user-images.githubusercontent.com/43363595/204237348-3d13827d-4537-4e73-a229-25b261834f01.png)

Reply:
I will have to take a deeper look at that. Right now all calculations are based on the point cloud. 

Isn't the z value directly proportional to disparity? Why not use just the z value of the points in the point cloud since they are already in the correct units (meters instead of pixels)?

Reply:
> Isn't the z value directly proportional to disparity?
it is inversely proportional. 
Might be fine. but would be easier to do a comparison was what I was thinking.